### PR TITLE
[otbn/otp_ctrl] Replicate dmem scrambling keystream

### PR DESCRIPTION
**Pull request by msfschaffner**
This replicates the keystream instead of instantiating multiple PRINCE primitives, as discussed in detail in #7054. This effectively means that the OTBN nonce is now only 64bit wide, and this patch changes the widths throughout the `otp_ctrl` and `otbn` designs.

Signed-off-by: Michael Schaffner <msf@opentitan.org>

### CODE REVIEW

**Comment by msfschaffner** [comment]
I chose to not merge the OTBN key request channel into the SRAM key request channel vector since OTBN does not have the same nonce width as the SRAMs (64bit instead of 128bit) since OTBN scrambling does not make use of the LFSR init feature.

Also, it turns out that just adjusting the widths is actually straightforward and requires less DV and toplevel changes.

File: hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc
Comment (rswarbrick) [inline hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc:22]: Would you mind making this an argument to the `ScrambledEcc32MemArea` constructor? I don't think the result would be more complicated, and the "else" blocks below end up as dead code otherwise.

**Review by rswarbrick — COMMENTED** [review]
Minor nit about configurability, but this looks good to me. I'll leave it to Greg to give a final review since this was his code originally.

**Review by GregAC — APPROVED** [review]
LGTM thanks for sorting it out

File: hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc
Comment (msfschaffner) [reply hw/dv/verilator/cpp/scrambled_ecc32_mem_area.cc]: No problem - moved into the constructor.

**Comment by msfschaffner** [comment]
Note: this does not significantly change the OTP design. On the DV side, the only change needed is a parameter change (see @cindychip's commit).

**Review by cindychip — APPROVED** [review]
LGTM

**Pull request merged**
